processing/src/main/java/org/apache/carbondata/processing/merger/CarbonCompactionUtil.java

```diff
@@ -460,6 +473,71 @@ public static boolean isRestructured(List<DataFileFooter> listMetadata,
     return false;
   }
 
+  public static RangeValues[] getRangesFromVals(Object[] tempRanges, Object[] startAndEnd) {
+    RangeValues[] ranges = new RangeValues[tempRanges.length + 1];
+    Object lastStart = startAndEnd[0];
+    for (int i = 0; i < tempRanges.length; i++) {
+      ranges[i] = new RangeValues(lastStart, tempRanges[i]);
+      lastStart = tempRanges[i];
+    }
+    ranges[tempRanges.length] = new RangeValues(lastStart, startAndEnd[1]);
+    return ranges;
+  }
+
+  public static Expression getAndExpressionForRange(String colName, int taskId, byte[] minVal,
+      byte[] maxVal, DataType dataType) {
+    Expression finalExpr;
+    Expression exp1, exp2;
+    Object currMinVal = null, currMaxVal = null;
+    if (dataType == DataTypes.STRING) {
```

Comment:
Max value can also be null when complete data is null, can u please check and add a test case for the same

Reply:
Added the testcase for it.